framework/src/parser/MooseSyntax.C

```diff
@@ -62,12 +63,16 @@ associateSyntax(Syntax & syntax, ActionFactory & action_factory)
 
   /// Variable/AuxVariable Actions
   syntax.registerActionSyntax("AddVariableAction", "Variables/*");
+  syntax.registerSyntaxType("Variables/*", "VariableName");
+  syntax.registerSyntaxType("Variables/*", "NonlinearVariableName");
```

Comment:
I am not sure what you mean in your first comment. You mean associating any variable with a block without iterating over all the allowed types?